app/controllers/cars_controller.rb

```diff
@@ -1,9 +1,19 @@
 class CarsController < ApplicationController
   before_action :set_car, only: %i[ show edit update destroy ]
 
+SORTING = {
+  'date_added_asc' => {date_added: :asc},
+  'date_added_desc' => {date_added: :desc},
+  'price_asc' => {price: :asc},
+  'price_desc' => {price: :desc}
+}
+
+
   # GET /cars or /cars.json
   def index
     @cars = Car.all.page params[:page]
+    @cars_total = Car.all
+    @cars = @cars.order(**SORTING[params[:sort_by]]) if params[:sort_by].present?
```

Comment:
Please use private def for this sorting and then use
def_name if params[:sort_by].present?